firmware/CMakeLists.txt

```diff
@@ -71,14 +71,45 @@ PRIVATE
 	${CMAKE_CURRENT_SOURCE_DIR}  # provides app.h
 )
 
+target_include_directories(os
+PRIVATE
+	${CMAKE_CURRENT_SOURCE_DIR}  # provides app.h
+)
+
 target_link_libraries(main
 PRIVATE
 	bindings
 )
 
+target_link_libraries(main
+PRIVATE
+	os
```

Comment:
By linking the os library in the top CMake, all projects must provide an os implementation. This breaks the ability to run windows-only projects like DemoMapper. The os should be added in the project's CMakeLists

Reply:
wrapped in IF statement